bin_heap.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+
+class BinaryHeap(object):
+    def __init__(self):
+        self.heaplist = [0]
+        self.currentsize = 0
```

Comment:
I would love to see you work out how to make this work without carrying the weird placeholder to allow 1-based indexing.  For one thing, it's non-pythonic.  For another, it prevents you including values less than 0 (and restricts you to numeric contents, rather than any self-compatible object).  Finally, it forces you to carry an extra instance attribute that should be required (`currentsize`)